### PR TITLE
Implemented HF designs for Search, (and other fixes)

### PR DESCRIPTION
**Pull request by rs929**
## Overview

- Implemented HF designs for search page
- Fixed Google Place location filter to only include drivable locations

## Changes Made

- Implemented HF designs for SearchRidesViewController
- Created new textField class ImageTextField
- Updated Scoop logo to Scooped
- Added filter to GMSPlace search to only include places in US and Canada
- Added custom designs for error messages
- Updated network call for Search endpoint to include GMSPlace location IDs

## Screenshots (delete if not applicable)

Before - After

</details>

### CODE REVIEW

File: Scoop/SearchRidesViewController.swift [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+//
+//  Created by Richie Sun on 4/7/23.
+//
+
+import GooglePlaces
+import UIKit
+
+class SearchRidesViewController: UIViewController {
+    
+    //MARK: Views
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift]: Add a space after `//`

File: Scoop/SearchRidesViewController.swift [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        calendarIcon.contentMode = .scaleAspectFit
+        view.addSubview(calendarIcon)
+        
+        calendarIcon.snp.makeConstraints { make in
+            make.centerY.equalTo(departureDateTextField)
+            make.trailing.equalTo(departureDateTextField).inset(12)
+            make.size.equalTo(24)
+        }
+        
+        [departureTextField, arrivalTextField, departureDateTextField].forEach { text in
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift:136]: 😻

File: Scoop/SearchRidesViewController.swift [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+                  self.presentErrorAlert(title: "Error", message: "Please complete all fields.")
+                  return
+              }
+        
+        NetworkManager.shared.searchLocation(depatureDate: tripDate, startLocation: departureID, endLocation: arrivalID) { response in
+            switch response {
+            case .success(let response):
+                self.navigationController?.pushViewController(MatchesViewController(rides: response.rides), animated: true)
+            case .failure(let error):
+                print(error)
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift]: Perhaps add some message before the error. For example, `print("Error in SearchRidesVC: "\(error.localizedDescription)"`. Very helpful when debugging to know where the error occurs!

File: Scoop/SearchRidesViewController.swift [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+            }
+        }
+    }
+
+    /** Converts a Date object into String format, where this one is specifically in the YYYY-MM-dd format. */
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift]: Nice documentation! Personally I use `///` for a single-line doc and `/** */` for multi-line but that's just me.

File: Scoop/SearchRidesViewController.swift [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    private func formatDate(dateToConvert: Date) -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "YYYY-MM-dd"
+        return dateFormatter.string(from: dateToConvert)
+    }
+    
+}
+
+//MARK: - UITextFieldDelegate
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift]: Add space after `//`

File: Scoop/SearchRidesViewController.swift [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+extension SearchRidesViewController: SearchInitialViewControllerDelegate {
+
+    func didSelectLocation(viewController: UIViewController, location: GMSPlace) {
+        if viewController is DepartureSearchViewController {
+            departurePlace = location
+            departureTextField.textField.text = location.name
+        } else if viewController is ArrivalSearchViewController {
+            arrivalPlace = location
+            arrivalTextField.textField.text = location.name
+
```
Comment (vinnie4k) [inline Scoop/SearchRidesViewController.swift]: Remove new line

**Review by vinnie4k — APPROVED** [review]
Your `SearchRidesViewController` class is very well written! Easy to follow, logic makes sense, perfect spacing, great documentation usage, etc. Keep up the good work Richie!

File: Scoop/SearchRidesViewController.swift [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+        }
+        
+        let solidline = UIView()
+        solidline.backgroundColor = .black
+        view.addSubview(solidline)
+        solidline.snp.makeConstraints { make in
+            make.height.equalTo(1)
+            make.width.equalTo(screenSize.width*solidLineMultiplier)
+            make.trailing.equalToSuperview()
+            make.top.equalTo(view.safeAreaLayoutGuide).inset(solidLineVerticalInset)
```
Comment (tiffany-pann) [inline Scoop/SearchRidesViewController.swift:67]: i can't tell if this indentation looks weird or if my github is just bugging out

File: Scoop/SearchRidesViewController.swift (hunk 2 above)
Comment (tiffany-pann) [reply Scoop/SearchRidesViewController.swift:136]: 😻

**Review by tiffany-pann — APPROVED** [review]
wow absolutely amazing work richie 🤩

File: Scoop/Controller/Home/HomeViewController.swift [hunk 8]
```diff
@@ -58,7 +58,7 @@ class HomeViewController: UIViewController {
         view.addSubview(headerView)
         
         headerView.snp.makeConstraints { make in
-            make.leading.trailing.equalToSuperview().inset(20)
+            make.leading.equalToSuperview()
             make.height.equalTo(90)
             make.top.equalTo(view.safeAreaLayoutGuide)
```
Comment (ReadeP) [inline Scoop/Controller/Home/HomeViewController.swift:63]: I think we should be setting a trailing anchor here, or at least defining some width for this view

File: Scoop/SearchRidesViewController.swift [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+import GooglePlaces
+import UIKit
+
+class SearchRidesViewController: UIViewController {
+    
+    //MARK: Views
+    private let arrivalLabel = UILabel()
+    private let arrivalTextField = ImageTextField()
+    private let calendarIcon = UIImageView()
```
Comment (ReadeP) [inline Scoop/SearchRidesViewController.swift]: `calenderIconImageView`

File: Scoop/SearchRidesViewController.swift [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+    
+    private func setupHeader() {
+        let dottedLineMultiplier = 0.52
+        let solidLineVerticalInset = -12.75
+        let solidLineMultiplier = 0.32
+        let screenSize = UIScreen.main.bounds
+        let dottedline = UIImageView(image: UIImage(named: "dottedline"))
+        
+        dottedline.contentMode = .scaleAspectFit
+        view.addSubview(dottedline)
```
Comment (ReadeP) [inline Scoop/SearchRidesViewController.swift:50]: I like to add a new line after addSubview and setting constraints. Makes it more readable imo

File: Scoop/SearchRidesViewController.swift [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+        departureDateLabel.isHidden = false
+    }
+    
+    @objc private func presentMatches() {
+        guard let departure = self.departureTextField.textField.text, !departure.isEmpty,
+              let arrival = self.arrivalTextField.textField.text, !arrival.isEmpty,
+              let date = self.departureDateTextField.text, !date.isEmpty,
+              let departureID = departurePlace?.placeID,
+              let arrivalID = arrivalPlace?.placeID else {
+                  self.presentErrorAlert(title: "Error", message: "Please complete all fields.")
```
Comment (ReadeP) [inline Scoop/SearchRidesViewController.swift]: You don't need `self` here

File: Scoop/SearchRidesViewController.swift [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+                  return
+              }
+        
+        NetworkManager.shared.searchLocation(depatureDate: tripDate, startLocation: departureID, endLocation: arrivalID) { response in
+            switch response {
+            case .success(let response):
+                self.navigationController?.pushViewController(MatchesViewController(rides: response.rides), animated: true)
```
Comment (ReadeP) [inline Scoop/SearchRidesViewController.swift]: Use a weak instance of self here `arrivalID) { [weak self] in ` to avoid potential retain cycle

Comment (ReadeP) [reply Scoop/SearchRidesViewController.swift]: You can then unwrap self using a guard let

File: Scoop/Supporting/Utilities/Extension+UIViewController.swift [hunk 13]
```diff
@@ -13,6 +13,8 @@ extension UIViewController {
     func presentErrorAlert(title: String, message: String) {
         let alertVC = UIAlertController(title: title, message: message, preferredStyle: .alert)
         alertVC.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+        alertVC.view.tintColor = .scoopDarkGreen
+        alertVC.title
```
Comment (ReadeP) [inline Scoop/Supporting/Utilities/Extension+UIViewController.swift]: What is this line?

**Review by ReadeP — APPROVED** [review]
Looks good! Just a few comments

**Pull request merged**